web/frontend/src/components/BaseMap.vue

```diff
@@ -122,6 +122,22 @@ export function setBottomCardAllowance(pixels?: number) {
   map?.resize();
 }
 
+/**
+ * Polyfill for geolocation permission
+ */
+async function geolocationPermissionState(): string {
+  if (navigator.permissions === undefined) {
+    // assume "unknown" on platforms like Safari 15 which don't
+    // support the permissions API.
+    return 'prompt';
```

Comment:
It's kind of unfortunate that we always assume "prompt". I think the only behavioral downside at this point is that on legacy safari we won't automatically highlight the users location upon subsequent visits. They'll have to click the button manually every time. 

Alternatives considered:

Always assume "granted" which would let subsequent visitors see their position, but it means that first time visitors would be immediately slapped with the location request which seems worse.

Another fancier option would be to use localStorage to record if a user had *ever* granted permission and return "granted" from that point, but it seemed like more complexity than it was worth, and it could get out of sync.

Reply:
Just so I'm understanding properly, this issue will go away in the future as more devices get onto Safari 16 right? As long as it goes away over time I'd prefer to do the first option of the ones you listed. I do NOT want to prompt every time :) Too many sites do that.